### PR TITLE
Enabling sampling of the supported upgrade tests

### PR DESCRIPTION
**Pull request by bradmwilliams**
This logic will scale down the number of support upgrade tests, performed by the release-controller, for `Stable` payloads by taking a sample of the upgrades as follows:
Currently, for 4.10.31, there are 55 `previous` upgrades:
``` bash
$ oc adm release info quay.io/openshift-release-dev/ocp-release:4.10.31-x86_64 -o json | jq '.metadata.previous'
[
  "4.10.10",
  "4.10.11",
  "4.10.12",
  "4.10.13",
  "4.10.14",
  "4.10.15",
  "4.10.16",
  "4.10.17",
  "4.10.18",
  "4.10.20",
  "4.10.21",
  "4.10.22",
  "4.10.23",
  "4.10.24",
  "4.10.25",
  "4.10.26",
  "4.10.27",
  "4.10.28",
  "4.10.29",
  "4.10.3",
  "4.10.30",
  "4.10.4",
  "4.10.5",
  "4.10.6",
  "4.10.7",
  "4.10.8",
  "4.10.9",
  "4.9.19",
  "4.9.21",
  "4.9.22",
  "4.9.23",
  "4.9.24",
  "4.9.25",
  "4.9.26",
  "4.9.27",
  "4.9.28",
  "4.9.29",
  "4.9.30",
  "4.9.31",
  "4.9.32",
  "4.9.33",
  "4.9.34",
  "4.9.35",
  "4.9.36",
  "4.9.37",
  "4.9.38",
  "4.9.39",
  "4.9.40",
  "4.9.41",
  "4.9.42",
  "4.9.43",
  "4.9.44",
  "4.9.45",
  "4.9.46",
  "4.9.47"
]
```
Instead of running 55 jobs, this logic will return the following (with the default sample size of 3):
4.Y-1.z -> 4.9
First:
```
    "4.9.47",
    "4.9.46",
    "4.9.45",
```
Last:
```
    "4.9.22",
    "4.9.21",
    "4.9.19",
```
Random selection:
```
    "4.9.42",
    "4.9.44",
    "4.9.34",
```
4.Y.Z -> 4.10
First:
```
    "4.10.30",
    "4.10.29",
    "4.10.28",
```
Last:
```
    "4.10.5",
    "4.10.4",
    "4.10.3",
```
Random selection:
```
    "4.10.9",
    "4.10.6",
    "4.10.26"
```
Basically, with a sample size of 3, the release-controller would only produce 18 jobs per release.

### CODE REVIEW

File: cmd/release-controller/sync_upgrade.go [hunk 1]
```diff
@@ -130,3 +140,87 @@ func getUpgradeProwJobs(release *releasecontroller.Release) ([]string, map[strin
...
+	for _, v := range supportedUpgrades {
+		if version, err := semver.Parse(v); err == nil {
+			releaseName := fmt.Sprintf("%d.%02d", version.Major, version.Minor)
+			if releaseBuckets[releaseName] == nil {
+				releaseBuckets[releaseName] = SemanticVersions{version}
+			} else {
+				releaseBuckets[releaseName] = append(releaseBuckets[releaseName], version)
+			}
+		} else {
+			fmt.Println(fmt.Sprintf("Unable to parse upgrade version: %q", version))
```
Comment (AlexNPavel) [inline cmd/release-controller/sync_upgrade.go:212]: We should log this as an error

**Review by AlexNPavel — CHANGES_REQUESTED** [review]
Just one nit about logging. Other than that this looks great.

File: cmd/release-controller/sync_upgrade.go (hunk 1 above)
Comment (bradmwilliams) [reply cmd/release-controller/sync_upgrade.go]: Done!
